src/main/java/Nimbus/CommandParser.java

```diff
@@ -0,0 +1,97 @@
+package Nimbus;
+
+public class CommandParser {
+    public static String[] validCommands = {"bye", "list", "mark", "unmark", "todo", "deadline", "event", "-help", "delete", "save"};
+    private String commandWord;
+    private String arguments;
+
+    public CommandParser(String input, int range) throws NimbusException {
```

Comment:
This constructor is >30 LOC which is a lot. Consider refactoring some of the inner nested levels of code to a different method.